packages/evm-development/source/deployer.ts

```diff
@@ -0,0 +1,70 @@
+import { inject, injectable } from "@mainsail/container";
+import { Contracts, Identifiers } from "@mainsail/contracts";
+import { Bindings } from "@mainsail/evm";
+import { Utils } from "@mainsail/kernel";
+import { ethers } from "ethers";
+
+import { ERC20 } from "./contracts.ts";
+
+@injectable()
+export class Deployer {
+	@inject(Identifiers.Application.Instance)
+	private readonly app!: Contracts.Kernel.Application;
+
+	@inject(Identifiers.Services.Log.Service)
+	private readonly logger!: Contracts.Kernel.Logger;
+
+	@inject(Identifiers.Evm.Instance)
+	private readonly evm!: Bindings.Evm;
+
+	@inject(Identifiers.Cryptography.Identity.Address.Factory)
+	private readonly addressFactory!: Contracts.Crypto.AddressFactory;
+
+	#genesisAddress!: string;
+
+	public async deploy(): Promise<void> {
+		const genesisBlock = this.app.config<Contracts.Crypto.CommitJson>("crypto.genesisBlock");
+		Utils.assert.defined(genesisBlock);
+
+		this.#genesisAddress = await this.addressFactory.fromPublicKey(genesisBlock.block.generatorPublicKey);
+
+		const result = await this.evm.transact({
+			caller: this.#genesisAddress,
+			data: Buffer.from(ethers.getBytes(ERC20.abi.bytecode)),
+		});
+
+		if (!result.success) {
+			throw new Error("failed to deploy erc20 contract");
+		}
+
+		this.logger.info(
+			`Deployed ERC20 dummy contract from ${this.#genesisAddress} to ${result.deployedContractAddress}`,
+		);
+
+		await this.ensureFunds(result.deployedContractAddress!);
+	}
+
+	private async ensureFunds(erc20ContractAddress: string): Promise<void> {
+		const secrets = this.app.config("validators.secrets");
+		Utils.assert.defined<string[]>(secrets);
+
+		const iface = new ethers.Interface(ERC20.abi.abi);
+		const amount = ethers.parseEther("1000");
+
+		for (const secret of secrets) {
```

Comment:
Use addresses from genesis block, because secrets can me modified and can be different on the different nodes.